components/scream/src/physics/p3/p3_main_impl.hpp

```diff
@@ -799,12 +799,14 @@ ::p3_main_part3(
       get_cloud_dsd2(qc_incld, nc_incld, mu_c(k), rho(k), nu(k), dnu, lamc(k), ignore1, ignore2, qc_gt_small);
 
       nc(k).set(qc_gt_small,nc_incld*cld_frac_l(k)); //cld_dsd2 might have changed incld nc... need consistency.
```

Comment:
There are two chunks of code here that depend on two different masks. Setting up two `if (mask.any())` blocks would be natural and increase performance. As a side effect, it would take care of the /0 case.